js/ai/src/resource.ts

```diff
@@ -54,7 +59,7 @@ export interface ResourceOptions {
  * A function that returns parts for a given resource.
  */
 export type ResourceFn = (
-  vars: Record<string, string>,
+  input: string,
```

Comment:
let's make input `{uri: string}` instead of just `string` so we leave ourselves options to enhance in the future

Reply:
Done.